lang/java/reef-common/src/main/java/org/apache/reef/driver/evaluator/EvaluatorRequest.java

```diff
@@ -96,6 +100,15 @@ public static Builder newBuilder(final EvaluatorRequest request) {
   }
 
   /**
+   * Access the node labels requested.
+   *
+   * @return the node labels requested.
+   */
+  public Map<String, String> getNodeLabels() {
+    return this.nodeLabels;
```

Comment:
What if the receiver of this instance starts to change it? After all, a `Map` is mutable and we return a reference here.